src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -239,6 +239,18 @@
     <value>Prefer 'static readonly' fields over constant array arguments if the called method is called repeatedly and is not mutating the passed array</value>
     <comment>{Locked="static readonly"}</comment>
   </data>
+  <data name="AvoidUnreliableStreamReadTitle" xml:space="preserve">
+    <value>Avoid unreliable call to 'Stream.Read'</value>
+  </data>
+  <data name="AvoidUnreliableStreamReadCodeFixTitle" xml:space="preserve">
+    <value>Use 'Stream.ReadExactly'</value>
+  </data>
+  <data name="AvoidUnreliableStreamReadDescription" xml:space="preserve">
+    <value>A call to 'Stream.Read' may return fewer bytes than requested, resulting in unreliable code if the return value is not checked.</value>
+  </data>
+  <data name="AvoidUnreliableStreamReadMessage" xml:space="preserve">
+    <value>Avoid unreliable call to '{0}'</value>
```

Comment:
Maybe something like, but if other OK with this message, I am fine. The complete message would look like:
`Avoid inexact read with 'System.IO.Ports.SerialStream.Read(byte[], int, int)' `, then we would suggest the fix `Use 'Stream.ReadExactly'` how is it look @gewarren  
```suggestion
    <value>Avoid inexact read with '{0}'</value>
```

Reply:
Thanks, I've added the new title and message in 254f3a868989ada9bc710a0d873e0efd5b5331a7.